Content.Server/Delivery/DeliverySystem.cs

```diff
@@ -73,10 +79,74 @@ protected override void GrantSpesoReward(Entity<DeliveryComponent?> ent)
         if (!TryComp<StationBankAccountComponent>(ent.Comp.RecipientStation, out var account))
             return;
 
+        var stationAccountEnt = (ent.Comp.RecipientStation.Value, account);
+
+        var multiplier = GetDeliveryMultiplier(ent!); // Resolve so we know it's got the component
+
+        _cargo.UpdateBankAccount(
+            stationAccountEnt,
+            (int)(ent.Comp.BaseSpesoReward * multiplier),
+           _cargo.CreateAccountDistribution((ent.Comp.RecipientStation.Value, account)));
+    }
+
+    /// <summary>
+    /// Runs the penalty logic: Announcing the penalty and calculating how much to charge the designated account
+    /// </summary>
+    /// <param name="ent">The delivery for which to run the penalty.</param>
+    /// <param name="reason">The penalty reason, displayed in front of the message.</param>
+    protected override void HandlePenalty(Entity<DeliveryComponent> ent, string? reason = null)
+    {
+        if (!TryComp<StationBankAccountComponent>(ent.Comp.RecipientStation, out var stationAccount))
+            return;
+
+        if (ent.Comp.WasPenalized)
+            return;
+
+        if (!_protoMan.TryIndex(ent.Comp.PenaltyBankAccount, out var accountInfo))
+            return;
+
+        var multiplier = GetDeliveryMultiplier(ent);
+
+        var localizedAccountName = Loc.GetString(accountInfo.Name);
+
+        reason ??= Loc.GetString(DefaultMessage);
+
+        var dist = new Dictionary<ProtoId<CargoAccountPrototype>, double>()
+        {
+            { ent.Comp.PenaltyBankAccount, 1.0 }
+        };
+
+        var penaltyAccountBalance = stationAccount.Accounts[ent.Comp.PenaltyBankAccount];
+        var calculatedPenalty = (int)(ent.Comp.BaseSpesoPenalty * multiplier);
+
+        // Prevents cargo from going into negatives
+        if (penaltyAccountBalance - calculatedPenalty < 0)
```

Comment:
```suggestion
        if (calculatedPenalty > penaltyAccountBalance )
```
same logic, but a little more readable and one calculation less